packages/concerto-core/lib/introspect/modelfile.js

```diff
@@ -750,9 +751,27 @@ class ModelFile extends Decorated {
                 this.importWildcardNamespaces.push(imp.namespace);
                 break;
             case `${MetaModelNamespace}.ImportTypes`:
-                imp.types.forEach( type => {
-                    this.importShortNames.set(type, `${imp.namespace}.${type}`);
-                });
+                if (this.getModelManager().isAliasedTypeEnabled()) {
+                    // map: alias name to the fqn's
+                    // imp.types and imp.aliasedTypes both are available
+                    const aliasedTypes = new Map();
+                    if (imp.aliasedTypes) {
+                        imp.aliasedTypes.forEach(({ name, aliasName }) => {
+                            if(ModelUtil.isPrimitiveType(aliasName)){
+                                throw new Error('Types cannot be aliased to primitive type');
+                            }
+                            aliasedTypes.set(name, aliasName);
+                        });
+                    }
+                    imp.types.forEach((type)=> aliasedTypes.has(type)? this.importShortNames.set(aliasedTypes.get(type),`${imp.namespace}.${type}`):this.importShortNames.set(type,`${imp.namespace}.${type}`));
```

Comment:
I think this line could use some whitespaces and some explanation
```suggestion
                    // `imp.types` include aliased imports too so
                    //  map aliased types to source FQN if it exists in `aliasedTypes` map
                    // we constructed above.
                    imp.types.forEach(
                        (type) => aliasedTypes.has(type)
                            ? this.importShortNames.set(aliasedTypes.get(type),`${imp.namespace}.${type}`)
                            : this.importShortNames.set(type,`${imp.namespace}.${type}`)
                    );
```

Reply:
Then you can remove the comment at line 754